[nn] Add remove_duplicate option for named_modules

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54928 [quant][graphmode][fx][refactor] Quantize by Use of a Tensor instead of Tensor
* #54924 [quant][graphmode][fx] Produce torch.cat instead of torch.ops.quantized.cat
* #54813 [quant][graphmode][fx] Optimize cat
* **#54812 [nn] Add remove_duplicate option for named_modules**

Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

jerryzh168 committed:
[nn] Add allow_duplicate option for named_modules

Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

albanD reviewed (commented):
The change sounds reasonable.
It is missing doc and test though.

Also we might want to consider "remove_duplicate=True" as a name?

jerryzh168 commented:
> The change sounds reasonable.
> It is missing doc and test though.
> 
> Also we might want to consider "remove_duplicate=True" as a name?

sure, I can change it to remove_duplicat, where do I add tests?

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add allow_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add remove_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

albanD reviewed (approved):
LGTM, thanks for the update and tests

jerryzh168 committed:
Update on "[nn] Add remove_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add remove_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add remove_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[nn] Add remove_duplicate option for named_modules"


Summary:
Needed for quantization since different attribute might refer to the same module instance

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27408376](https://our.internmc.facebook.com/intern/diff/D27408376)

[ghstack-poisoned]
